app/constants/autocomplete.ts

```diff
@@ -17,6 +17,10 @@ export const ALL_SEARCH_FLAGS_REGEX = /\b\w+:/g;
 
 export const CODE_REGEX = /(`+)([^`]|[^`][\s\S]*?[^`])\1(?!`)| *(`{3,}|~{3,})[ .]*(\S+)? *\n([\s\S]*?\s*)\3 *(?:\n+|$)/g;
 
+export const MENTIONS_REGEX = /(?:\B|\b_+)@([a-z0-9.\-_]+)/gi;
```

Comment:
will this regex work with non-latin characters? say Japanese, Cyrillic, etc..?

I guess it also differs from https://github.com/mattermost/mattermost-mobile/blob/main/app/helpers/api/user.ts#L9 which I think should be updated as well to match non-latin characters.

Same for https://github.com/mattermost/mattermost-mobile/blob/main/android/app/src/main/java/com/mattermost/helpers/push_notification/Post.kt#L19 and https://github.com/mattermost/mattermost-mobile/blob/main/ios/Gekidou/Sources/Gekidou/Networking/Network%2BPosts.swift#L156

Reply:
```ts
export const MENTIONS_REGEX = /(?:\B|\b_+)@([a-z0-9.\-_]+)/gi;
```
this one is the same as the webapp's

Reply:
Does it match non latin usernames? If the answer to that is no, then webapp needs fixing as well

Reply:
@enahum sorry I posted by mistake!! I wanted to add that I can add support for unicode, and yes we have to do the same for the webapp.

My first comment was about:

> I guess it also differs from https://github.com/mattermost/mattermost-mobile/blob/main/app/helpers/api/user.ts#L9 

and I wanted to ask if we need to change that to the MENTIONS_REGEX as well.

Reply:
Yes we need to change that one too